ui/src/utils/hocs/withModels.js

```diff
@@ -14,13 +14,15 @@ import {
   deleteModel,
   updateModel
 } from 'ui/redux/actions';
+import areEqualProps from './areEqualProps';
 
 export default (WrappedComponent) => {
   class WithModels extends Component {
-    componentWillMount = () => this.fetchModels(this.props)
+    componentDidMount = () => this.fetchModels(this.props);
 
-    componentWillReceiveProps = nextProps =>
-      this.fetchModels(nextProps);
+    shouldComponentUpdate = nextProps => !areEqualProps(this.props, nextProps);
```

Comment:
To avoid the following warning.

> Uncaught Invariant Violation: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loop

I expects this also improves the performance by reducing rendering and fetchModels.